<a href="https://colab.research.google.com/github/MarteDevs/estadistica-con-python-probabilidad-y-muestreo/blob/main/Inicial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***
# <font color=green size=10>CURSO DE ESTADÍSTICA - PARTE 2</font>
***

# <font color=green>1 CONOCIENDO LOS DATOS</font>
***

## <font color=green>1.1 Dataset del proyecto</font>
***

### Encuesta Nacional por Muestreo de Hogares - 2015

La **Encuesta Nacional por Muestreo de Hogares - ENMH** en Brasil anualmente investiga, de manera permanente, características generales de la población, educación, trabajo, ingresos, vivienda y otras, con frecuencia variables, de acuerdo a las necesidades de información del país, tales como las características de migración, fecundidad, nupcialidad, salud, seguridad alimentaria, entre otros temas. La recopilación de estas estadísticas constituye, durante los 49 años de realización de la investigación, un importante instrumento para la formulación, validación y evaluación de políticas orientadas al desarrollo socioeconómico y la mejora de las condiciones de vida en Brasil.

### Fuente de los Datos

Instituto Brasileño de Geografía y Estadística (IBGE) Encuesta Nacional por Muestreo de Hogares(PNAD) en el 2015

### Variables utilizadas

> ### Ingresos
> ***

Ingresos mensuales del trabajo principal para personas de 10 años o más.

> ### Edad
> ***

Edad del residente en años en la fecha de referencia .

> ### Altura (elaboración propia)
> ***

Altura del habitante en metros.

> ### UF
> ***

Unidad Federal

|Código|Descripción|
|---|---|
|11|Rondônia|
|12|Acre|
|13|Amazonas|
|14|Roraima|
|15|Pará|
|16|Amapá|
|17|Tocantins|
|21|Maranhão|
|22|Piauí|
|23|Ceará|
|24|Rio Grande do Norte|
|25|Paraíba|
|26|Pernambuco|
|27|Alagoas|
|28|Sergipe|
|29|Bahia|
|31|Minas Gerais|
|32|Espírito Santo|
|33|Rio de Janeiro|
|35|São Paulo|
|41|Paraná|
|42|Santa Catarina|
|43|Rio Grande do Sul|
|50|Mato Grosso do Sul|
|51|Mato Grosso|
|52|Goiás|
|53|Distrito Federal|

> ### Sexo
> ***

|Código|Descripción|
|---|---|
|0|Masculino|
|1|Femenino|

> ### Años de Estudio
> ***

|Código|Descripción|
|---|---|
|1|Sin educación o menos de 1 año|
|2|1 año|
|3|2 años|
|4|3 años|
|5|4 años|
|6|5 años|
|7|6 años|
|8|7 años|
|9|8 años|
|10|9 años|
|11|10 años|
|12|11 años|
|13|12 años|
|14|13 años|
|15|14 años|
|16|15 años o más|
|17|No determinado|
||No aplica|

> ### Color
> ***

|Código|Descripción|
|---|---|
|0|Indígena|
|2|Blanco|
|4|Negro|
|6|Amarillo|
|8|Pardo|
|9|Sin declarar|

#### <font color='red'>Nota</font>
***
> Se realizaron los siguientes tratamientos sobre los datos originales:
> 1. Se eliminaron los registros en los que **Ingreso** no era válido. (999 999 999 999);
> 2. Se eliminaron los registros en los que **Ingreso** eran missing;
> 3. Solo se consideraron los registros de **Personas de Referencia** de cada hogar (responsable del hogar).

### Importando pandas y leyendo el dataset del proyecto

https://pandas.pydata.org/

In [1]:
# se debe tener documentado el tratamiento de datos
import pandas as pd

In [2]:
datos = pd.read_csv('/content/datos.csv', encoding='latin1')

In [3]:
datos.head()

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
0      11     0    23      8               12      800  1.603808
1      11     1    23      2               12     1150  1.739790
2      11     1    35      8               15      880  1.760444
3      11     0    46      2                6     3500  1.783158
4      11     1    47      8                9      150  1.690631

---
---

# <font color=green>2 DISTRIBUCIONES DE PROBABILIDAD</font>
***

## <font color=red>Problema</font>
***

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide hacer la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

## <font color=green>2.1 Distribución Binomial</font>
***

Un evento **binomial** se caracteriza por la posibilidad de que solo ocurran dos categorías. Estas categorías juntas representan todo el espacio muestral, siendo también mutuamente excluyentes, es decir, la ocurrencia de una implica la no ocurrencia de la otra.

En el análisis estadístico, el uso más común de la distribución binomial es para resolver problemas que involucran situaciones de **éxito** y **fracaso**.

# $$P(k)=\binom{n}{k} p^k q^{n-k}$$

Donde:

$p$ = probabilidad de éxito

$q = (1 - p)$ = probabilidad de fracaso

$n$ = número de eventos estudiados

$k$ = número de eventos deseados que tienen éxito

### Experimento Binomial

1. Realización de $n$ pruebas idénticas.

2. Las pruebas son independientes.

3. Solo son posibles dos resultados, por ejemplo: Verdadero o falso; Cara o cruz; Éxito o fracaso.

4. La probabilidad de éxito está representada por $p$ y la probabilidad de fracaso por $1-p = q$. Estas probabilidades no cambian de una prueba para otra.

### Media de la distribución binomial

El valor esperado o la media de la distribución binomial es igual al número de experimentos realizados multiplicado por la probabilidad de que ocurra el evento.

# $$\mu = n \times p$$

### Desviación estándar de la distribución binomial

La desviación estándar es la raíz quadrada del producto entre el número de experimentos, la probabilidad de éxito y la probabilidad de fracaso.

# $$\sigma = \sqrt{n \times p \times q}$$

### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.comb.html

In [4]:
from scipy.special import comb
# esta libreria nos ayuda trabajar

### Combinaciones

El número de combinaciones de $n$ objetos, tomando $k$ cada vez, es:

# $$C_{k}^{n} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$

Donde

## $$n! = n\times(n-1)\times(n-2)\times...\times(2)\times(1)$$
## $$k! = k\times(k-1)\times(k-2)\times...\times(2)\times(1)$$

Por definición

## $$0! = 1$$

## <font color='blue'>Ejemplo: Mega Sena</font>

En una rueda de lotería Mega Sena tenemos un total de **60 números** para elegir, donde la apuesta mínima es de **seis números**. Usted que es curioso(a) decide calcular la probabilidad de ganar la Mega Sena con solo **un juego**. Para eso necesitamos saber cuántas **combinaciones de seis números se pueden formar con los 60 números disponibles**.

### $$C_{6}^{60} = \binom{60}{6} = \frac{60!}{6!(60 - 6)!}$$

In [5]:
combinaciones = comb(60, 6)
print(f'El número de combinaciones es: {combinaciones}')
# numero de posibilidades que tengo de escoger nuemreos ganadores

El número de combinaciones es: 50063860.0


In [6]:
# la probalididad de ganar es
probabilidad = 1 / combinaciones
print(f'La probabilidad de ganar es: {probabilidad:0.15f}')


La probabilidad de ganar es: 0.000000019974489


## **RETO**

In [7]:
combinaciones_2 = comb(25, 20)
print(f'El número de combinaciones es: {combinaciones_2:0.15f}')

El número de combinaciones es: 53130.000000000000000


In [8]:
probabilidad_2= 1/combinaciones_2
print(f'La probabilidad de ganar es: {probabilidad_2:0.15f}')

La probabilidad de ganar es: 0.000018821757952


## <font color='blue'>Ejemplo: Concurso para científico de datos</font>

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide tomar la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

### ¿Cuál es la cantidad de pruebas ($n$)?

In [9]:
n = 10
n

10

### ¿Las pruebas son independientes?

Sí. La opción elegida en una pregunta no influye en la opción elegida en otra pregunta.

### ¿Sólo dos resultados son posibles en cada prueba?

Sí. El candidato tiene dos posibilidades, ACERTAR o FALLAR una pregunta.

### ¿Cuál es la probabilidad de éxito ($p$)?

In [10]:
numero_de_alternativas_por_pregunta = 3
p = 1 / numero_de_alternativas_por_pregunta
p

0.3333333333333333

### ¿Cuál es la probabilidad de fracaso ($q$)?

In [11]:
# probabilidad de fracaso
q = 1 - p
q

0.6666666666666667

### ¿Cuál es el número total de eventos para tener éxito ($k$)?

In [12]:
k = 5
k

5

### Solución 1

In [13]:
# Se calcula la probabilidad de obtener exactamente k=5 aciertos en n=10 intentos
# utilizando la fórmula de la distribución binomial: P(k) = C(n, k) * p^k * q^(n-k)
# donde C(n, k) es el número de combinaciones de n en k (calculado con comb(n, k)),
# p es la probabilidad de éxito en un solo intento, y q es la probabilidad de fracaso.
probabilidad_de_acierto = (comb(10,5) * (p ** k)* (q**(n-k)))
# Se imprime el resultado de la probabilidad de acierto con formato a 8 decimales.
print(f'La probabilidad de acierto es: {probabilidad_de_acierto:0.8f}')

La probabilidad de acierto es: 0.13656455


### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom.html

In [14]:
from scipy.stats import binom

### Solución 2

In [15]:
# Se calcula la probabilidad de obtener exactamente k aciertos en n intentos,
# utilizando la función de masa de probabilidad (pmf) de la distribución binomial
# de la librería scipy.stats.
# k: número de eventos deseados que tienen éxito (en este caso, 5 aciertos)
# n: número de eventos estudiados (en este caso, 10 preguntas)
# p: probabilidad de éxito en un solo intento (en este caso, 1/3 de probabilidad de adivinar correctamente)
probabilidad = binom.pmf(k, n, p)
# Se imprime el resultado de la probabilidad de acierto con formato a 8 decimales.
print(f'La probabilidad de acierto es: {probabilidad:0.8f}')

La probabilidad de acierto es: 0.13656455


### Obtenga la probabilidad de que el candidato apruebe

### $$P(acertar \geq 5) = P(5) + P(6) + P(7) + P(8) + P(9) + P(10)$$

In [16]:
# forma mamual
# probabilidad de acertar minimo 5 preguntas o a pruebe
binom.pmf(5,n,p) + binom.pmf(6,n,p) + binom.pmf(7,n,p) + binom.pmf(8,n,p) + binom.pmf(9,n,p) + binom.pmf(10,n,p)

np.float64(0.21312808006909476)

In [17]:
# forma 2 automatizadada
binom.pmf([5,6,7,8,9,10],n,p).sum()
# creamos una lista con los datos

np.float64(0.21312808006909476)

In [18]:
# forma 3 forma mas simple
# Calcula la probabilidad de obtener a lo sumo 4 aciertos (es decir, 0, 1, 2, 3 o 4 aciertos)
# utilizando la función de distribución acumulada (cdf) de la distribución binomial
# con n=10 ensayos y probabilidad de éxito p.
# Esta es la probabilidad de NO aprobar, ya que se necesita al menos 5 aciertos para aprobar.
probabilidad_hasta_4 = binom.cdf(4, n, p)
# Muestra la probabilidad de aprobar.
print(probabilidad_hasta_4)

0.7868719199309048


In [19]:
# Calcula la probabilidad de obtener al menos 5 aciertos (es decir, 5, 6, 7, 8, 9 o 10 aciertos).
# Esto se hace restando la probabilidad de obtener 4 o menos aciertos (probabilidad_hasta_4) de 1.
# Esta es la probabilidad de aprobar el examen.
probabilidad_aprobar = 1 - probabilidad_hasta_4
print(probabilidad_aprobar)

0.21312808006909523


In [20]:
# forma mas simple (función de supervivencia)
# Calcula la probabilidad de obtener más de k=4 aciertos (es decir, 5, 6, 7, 8, 9 o 10 aciertos)
# utilizando la función de supervivencia (sf) de la distribución binomial
# con n=10 ensayos y probabilidad de éxito p.
# binom.sf(k, n, p) = P(X > k) = P(X >= k+1)
binom.sf(4,n,p)
# esta funcion hace diretamente la funcion la probabilidad de acertar

np.float64(0.21312808006909525)

## RETO

In [21]:
# from scipy.stats import binom

p_mon= 1 / 2  # Probabilidad de salir CRUZ
n_mon = 4          # Total de lanzamientos
k_mon = 2          # Total de aciertos (CRUZ hacia arriba)

binom.pmf(k_mon, n_mon, p_mon)

np.float64(0.375)

In [22]:
# from scipy.stats import binom

p = 1 / 6   # Probabilidad de salir el número CINCO
n = 10      # Total de lanzamientos

print(f'{binom.sf(2, n, p):.2%}')

22.48%


## <font color='blue'>Ejemplo: Yincana</font>

Una ciudad del interior realiza una yincana cada año para recaudar fondos para el hospital de la ciudad. En la última competencia se sabe que la **proporción de mujeres participantes fue del 60%**. **El número total de equipos, con 12 miembros, inscritos en la yincana de este año es 30**. Con la información anterior, responda: ¿Cuántos equipos deberían estar formados por **8 mujeres**?

### Solución

In [23]:
p = 0.6
p
# probabilididad de selecioanr a una mujer

0.6

In [24]:
n = 12
n
# cantidad de personas en el equipo
# cantidad de veces de selecioanr una persona

12

In [25]:
k = 8
k
# evento de existo

8

In [26]:
probabilidad_yinkana = binom.pmf(k,n,p)
print(f'La probabilidad de yinkana es: {probabilidad_yinkana:0.8f}')
# esto pasa a ser nuestros exito

La probabilidad de yinkana es: 0.21284094


In [27]:
# De los 30 cuantso equipos pueden estar formados con esta configuracion
media_binomial = 30 * probabilidad_yinkana
print(f'La media de la distribución binomial es: {media_binomial:0.0f}')

La media de la distribución binomial es: 6


## RETO

In [28]:
# from scipy.stats import binom

# p: Probabilidad de éxito en un solo ensayo/evento.
# En un problema, busca la proporción o el porcentaje de un resultado específico.
p = 0.22

# n: Número de ensayos o repeticiones en un experimento binomial individual.
# Busca el número de veces que se repite el evento en cada "grupo" o "muestra".
n = 3

# k: Número deseado de éxitos en los n ensayos.
# Busca el número específico de veces que se espera que ocurra el evento de éxito.
k = 2

# N: Número total de grupos o experimentos que se están considerando.
# Busca la cantidad total de "muestras" o "equipos" en el problema.
N = 50

# Calcula la probabilidad de obtener exactamente k éxitos en n ensayos utilizando la función de masa de probabilidad (pmf) de la distribución binomial.
probabilidad = binom.pmf(k, n, p)

# Calcula la media de la distribución binomial (número esperado de éxitos en N grupos) multiplicando la probabilidad de éxito en un grupo por el número total de grupos.
media = probabilidad * N

# Imprime la probabilidad calculada con formato a 8 decimales.
print(f'La probabilidad de acierto es: {probabilidad:0.8f}')
# Imprime la media calculada con formato a 8 decimales.
print(f'La media de la distribución binomial es: {media:0.8f}')

La probabilidad de acierto es: 0.11325600
La media de la distribución binomial es: 5.66280000


---
---

## <font color=red>Problema</font>
***

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

## <font color=green>2.2 Distribución Poisson</font>
***

Se utiliza para describir el número de ocurrencias en un tiempo o espacio específico. Los eventos se caracterizan por la posibilidad de contar los éxitos, pero no poder contar los fracasos.

Como ejemplos de procesos donde podemos aplicar la distribución de Poisson tenemos la determinación del número de clientes que ingresan a una tienda en un momento determinado, el número de autos que llegan a un Auto-servicio de una cafetería a la hora del almuerzo, la determinación del número de accidentes registrados en un tramo de carretera, etc.

# $$P(k) = \frac{e^{-\mu}(\mu)^k}{k!}$$

Donde:

$e$ = constante cuyo valor aproximado es 2.718281828459045

$\mu$ = representa el número promedio de ocurrencias en un intervalo de tiempo o espacio dado

$k$ = número de éxitos en el intervalo deseado

### Experimento Poisson

1. La probabilidad de que ocurra es la misma en todo el intervalo observado.

2. El número de ocurrencias en un intervalo dado es independiente del número de ocurrencias en otros intervalos.

3. La probabilidad de que ocurra es la misma en intervalos de igual longitud.

### Media de la distribución Poisson

# $$\mu$$

### Desviación estándar de la distribución Poisson

# $$\sigma = \sqrt{\mu}$$

### Importando bibliotecas

http://www.numpy.org/

In [29]:
import numpy as np

In [30]:
# numero de Euler
np.e

2.718281828459045

## <font color='blue'>Ejemplo: Delivery</font>

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

### ¿Cuál es el número promedio de ocurrencias por hora ($\mu$)?

In [31]:
media = 20
media
# se recibe como media

20

### ¿Qué número de ocurrencias queremos obtener en el período ($k$)?

In [32]:
k = 15
k
# numero de ocurrencias que queremos obtener

15

### Solución 1

In [33]:
import math

In [34]:
probabilidad_delivery = ((np.e**(-media)) * (media**k))/(math.factorial(k))
print(f'La probabilidad de recibir 15 pedidos es: {probabilidad_delivery:0.8f}')

La probabilidad de recibir 15 pedidos es: 0.05164885


### Importando bibliotecas

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html

### Solución 2

In [35]:
# funcion mas obtimizada
from scipy.stats import poisson

In [36]:
# Calcula la probabilidad de obtener exactamente k ocurrencias (pedidos) en un intervalo de tiempo (por hora),
# utilizando la función de masa de probabilidad (pmf) de la distribución de Poisson
# de la librería scipy.stats.
# k: número de éxitos deseados en el intervalo (en este caso, 15 pedidos)
# media: número promedio de ocurrencias en el intervalo (en este caso, 20 pedidos por hora)
probabilidad_delivery = poisson.pmf(k, media)

# Se imprime el resultado de la probabilidad calculada con formato a 8 decimales.
print(f'La probabilidad de recibir 15 pedidos es: {probabilidad_delivery:0.8f}')

La probabilidad de recibir 15 pedidos es: 0.05164885


## **RETO**

In [37]:
media = 20
k = 25

In [38]:
probabilidad_delivery = poisson.pmf(k, media)

# Se imprime el resultado de la probabilidad calculada con formato a 8 decimales.
print(f'La probabilidad de recibir 15 pedidos es: {probabilidad_delivery:0.8f}')
# Se imprime el resultado de la probabilidad calculada como porcentaje.
print(f'La probabilidad de recibir 15 pedidos es: {probabilidad_delivery:.2%}')

La probabilidad de recibir 15 pedidos es: 0.04458765
La probabilidad de recibir 15 pedidos es: 4.46%


---
---

## <font color=red>Problema</font>
***

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, sea menor de 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, tenga entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, tenga más de 1.90 metros.


## <font color=green>2.3 Distribución Normal</font>
***

La distribución normal es una de las más utilizadas en estadística. Es una distribución continua, donde la distribución de frecuencias de una variable cuantitativa tiene forma de campana y es simétrica en relación con su media.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img001.png' width='500px'>

### Características importantes

1. Es simétrica alrededor de la media;

2. El área bajo la curva corresponde a la proporción 1 o 100%;

3. Las medidas de tendencia central (media, mediana y moda) tienen el mismo valor;

4. Los extremos de la curva tienden al infinito en ambas direcciones y, teóricamente, nunca tocan el eje $x$;

5. La desviación estándar define la planitud y el ancho de la distribución. Las curvas más anchas y planas tienen valores de desviación estándar más altos;

6. La distribución se define por su media y desviación estándar;

7. La probabilidad siempre será igual al área bajo la curva, delimitada por los límites superior e inferior.

# $$f(x) = \frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

La probabilidad se obtiene del área bajo la curva, delimitada por los límites superior e inferior especificados. Se puede ver un ejemplo en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img002.png' width='500px'>


Para obtener el área anterior, simplemente calcule la integral de la función para los intervalos determinados. Según la siguiente ecuación:

# $$P(L_i<x<L_s) = \int_{L_i}^{L_s}\frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

$L_i$ = límite inferior

$L_s$ = límite superior

### Tablas estandarizadas

Se crearon tablas estandarizadas para facilitar la obtención de los valores de las áreas bajo la curva normal y eliminar la necesidad de resolver integrales definidas.

Para ver los valores en una tabla estandarizada, simplemente transforme nuestra variable en una variable estandarizada $Z$.

Esta variable $Z$ representa la desviación en las desviaciones estándar de un valor de la variable original en relación con la media.

# $$Z = \frac{x-\mu}{\sigma}$$

Donde:

$x$ = variable normal con media $\mu$ y desviación estándar $\sigma$

$\sigma$ = desviación estándar

$\mu$ = media

### Construyendo la tabla normal estandarizada
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

In [40]:
import pandas as pd
import numpy as np
from scipy.stats import norm

# Creamos un DataFrame vacío para representar la tabla normal estandarizada.
# Los índices (filas) representan las unidades y las primeras decimales del valor Z.
# Las columnas representan la segunda decimal del valor Z.
tabla_normal_estandarizada = pd.DataFrame(
    [],
    index=["{0:0.2f}".format(i / 100) for i in range(0, 400, 10)],
    columns = ["{0:0.2f}".format(i / 100) for i in range(0, 10)])

# Llenamos el DataFrame con los valores del área bajo la curva normal estándar.
# Estos valores corresponden a la probabilidad acumulada hasta el valor Z dado.
for index in tabla_normal_estandarizada.index:
    for column in tabla_normal_estandarizada.columns:
        # Calculamos el valor Z combinando el índice de la fila y la columna.
        Z = np.round(float(index) + float(column), 2)
        # Obtenemos el área bajo la curva normal estándar hasta el valor Z utilizando la función cdf (función de distribución acumulada) de scipy.stats.norm.
        tabla_normal_estandarizada.loc[index, column] = "{0:0.4f}".format(norm.cdf(Z))

# Renombramos el eje de las columnas a 'Z' para mayor claridad.
tabla_normal_estandarizada.rename_axis('Z', axis = 'columns', inplace = True)

# Mostramos la tabla normal estandarizada generada.
# Esta tabla permite encontrar la probabilidad de que una variable aleatoria normal estándar sea menor o igual a un valor Z dado.
tabla_normal_estandarizada

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
0.00  0.5000  0.5040  0.5080  0.5120  0.5160  0.5199  0.5239  0.5279  0.5319   
0.10  0.5398  0.5438  0.5478  0.5517  0.5557  0.5596  0.5636  0.5675  0.5714   
0.20  0.5793  0.5832  0.5871  0.5910  0.5948  0.5987  0.6026  0.6064  0.6103   
0.30  0.6179  0.6217  0.6255  0.6293  0.6331  0.6368  0.6406  0.6443  0.6480   
0.40  0.6554  0.6591  0.6628  0.6664  0.6700  0.6736  0.6772  0.6808  0.6844   
0.50  0.6915  0.6950  0.6985  0.7019  0.7054  0.7088  0.7123  0.7157  0.7190   
0.60  0.7257  0.7291  0.7324  0.7357  0.7389  0.7422  0.7454  0.7486  0.7517   
0.70  0.7580  0.7611  0.7642  0.7673  0.7704  0.7734  0.7764  0.7794  0.7823   
0.80  0.7881  0.7910  0.7939  0.7967  0.7995  0.8023  0.8051  0.8078  0.8106   
0.90  0.8159  0.8186  0.8212  0.8238  0.8264  0.8289  0.8315  0.8340  0.8365   
1.00  0.8413  0.8438  0.8461  0.8485  0.8508  0.8531  0.8554  0.8577  0.8599   
1.10  0.8643  0.8665  0.8686  0.8708  0.8729  0.8749  0.8770  0.8790  0.8810   
1.20  0.8849  0.8869  0.8888  0.8907  0.8925  0.8944  0.8962  0.8980  0.8997   
1.30  0.9032  0.9049  0.9066  0.9082  0.9099  0.9115  0.9131  0.9147  0.9162   
1.40  0.9192  0.9207  0.9222  0.9236  0.9251  0.9265  0.9279  0.9292  0.9306   
1.50  0.9332  0.9345  0.9357  0.9370  0.9382  0.9394  0.9406  0.9418  0.9429   
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   
2.60  0.9953  0.9955  0.9956  0.9957  0.9959  0.9960  0.9961  0.9962  0.9963   
2.70  0.9965  0.9966  0.9967  0.9968  0.9969  0.9970  0.9971  0.9972  0.9973   
2.80  0.9974  0.9975  0.9976  0.9977  0.9977  0.9978  0.9979  0.9979  0.9980   
2.90  0.9981  0.9982  0.9982  0.9983  0.9984  0.9984  0.9985  0.9985  0.9986   
3.00  0.9987  0.9987  0.9987  0.9988  0.9988  0.9989  0.9989  0.9989  0.9990   
3.10  0.9990  0.9991  0.9991  0.9991  0.9992  0.9992  0.9992  0.9992  0.9993   
3.20  0.9993  0.9993  0.9994  0.9994  0.9994  0.9994  0.9994  0.9995  0.9995   
3.30  0.9995  0.9995  0.9995  0.9996  0.9996  0.9996  0.9996  0.9996  0.9996   
3.40  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997   
3.50  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998   
3.60  0.9998  0.9998  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.70  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.80  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.90  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000   

Z       0.09  
0.00  0.5359  
0.10  0.5753  
0.20  0.6141  
0.30  0.6517  
0.40  0.6879  
0.50  0.7224  
0.60  0.7549  
0.70  0.7852  
0.80  0.8133  
0.90  0.8389  
1.00  0.8621  
1.10  0.8830  
1.20  0.9015  
1.30  0.9177  
1.40  0.9319  
1.50  0.9441  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952  
2.60  0.9964  
2.70  0.9974  
2.80  0.9981  
2.90  0.9986  
3.00  0.9990  
3.10  0.9993  
3.20  0.9995  
3.30  0.9997  
3.40  0.9998  
3.50  0.9998  
3.60  0.9999  
3.70  0.9999  
3.80  0.9999  
3.90  1.0000

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img003.png' width='250px'>

La tabla anterior proporciona el área bajo la curva entre $-\infty$ y $Z$ desviaciones estándar por encima de la media. Recordando que dado que estos son valores estandarizados, tenemos $\mu = 0$.

## <font color='blue'>Ejemplo: ¿Cuánto mides?</font>

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, mida más de 1.90 metros.

### Problema A - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img004.png' width='350px'>


### Obtener la variable estandarizada $Z$

In [41]:
# construimos nuestra variable estandarizada
media = 1.70
media

1.7

In [43]:
desviacion_estandar = 0.1
desviacion_estandar
# desviacion estandar

0.1

In [46]:
Z = (1.8 - media)/ desviacion_estandar
Z
# calculamos el valor estandarizado

1.0000000000000009

### Solución 1 - Utilizando la tabla

In [49]:
probabilidad = 0.8413
probabilidad
# A. probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros.

0.8413

### Solución 2 - Utilizando Scipy

In [52]:
# funcion mas sencilla
from scipy.stats import norm

# Calcula la probabilidad de que una variable aleatoria normal estándar sea menor o igual a Z.
# Z es el valor estandarizado que calculamos anteriormente.
# La función norm.cdf(Z) devuelve el área bajo la curva normal estándar a la izquierda de Z,
# lo que representa la probabilidad de que una observación sea menor o igual a Z.
norm.cdf(Z)

np.float64(0.8413447460685431)

# **EJERCICIO**

In [57]:
media = 70
desviacion_estandar = 5
Z = (85 - media) / desviacion_estandar

norm.cdf(Z)

np.float64(0.9986501019683699)

### Problema B - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img005.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [62]:
media = 1.70
media

1.7

In [63]:
desviacion_estandar=0.1
desviacion_estandar

0.1

In [64]:
# obetenemos el area
Z = (1.7 - media) / desviacion_estandar
Z

0.0

In [65]:
Z = (1.8 - media) / desviacion_estandar
Z
# obetenemos el area para 1.80

1.0000000000000009

### Solución 1 - Utilizando la tabla

In [66]:
probabilidad = 0.8413
probabilidad

0.8413

In [69]:
probabilidad = (0.8413 - 0.5) * 2
probabilidad

0.6826000000000001

### Solución 2 - Utilizando Scipy

In [71]:
# cuando queremos saber la probabilidad entre un intervalo usamos esto
Z_superior = (1.8 - media) / desviacion_estandar
Z_superior

1.0000000000000009

In [72]:
probabilidad_superior = norm.cdf(Z_superior)
probabilidad_superior

np.float64(0.8413447460685431)

In [73]:
Z_inferior  = (1.6 - media) / desviacion_estandar
Z_inferior
# obetenemos el area para 1.80

-0.9999999999999987

In [74]:
probabilidad_inferior = norm.cdf(Z_inferior)
probabilidad_inferior

np.float64(0.1586552539314574)

In [75]:
probabilidad_total = probabilidad_superior - probabilidad_inferior
probabilidad_total

np.float64(0.6826894921370857)

## **EJERCICIO**

In [83]:
media = 300.00
desviacion_estandar = 50.00

In [84]:
x_superior = (250.00 - media) / desviacion_estandar
x_superior


-1.0

In [85]:
x_inferior = (400.00 - media) / desviacion_estandar
x_inferior

2.0

In [90]:
probabilidad_superior = norm.cdf(x_superior)
# Se imprime el resultado de la probabilidad superior como porcentaje.
print(f'La probabilidad superior es: {probabilidad_superior:.2%}')

La probabilidad superior es: 15.87%


In [91]:
probabilidad_inferior = norm.cdf(x_inferior)
# Se imprime el resultado de la probabilidad inferior como porcentaje.
print(f'La probabilidad inferior es: {probabilidad_inferior:.2%}')

La probabilidad inferior es: 97.72%


In [92]:
# Datos del problema
media = 300.00
desviacion_estandar = 50.00

# 1) Probabilidad entre $ 250,00 y $ 350,00

# Calcular los valores Z para los límites inferior y superior
Z_250 = (250.00 - media) / desviacion_estandar
Z_350 = (350.00 - media) / desviacion_estandar

# Calcular la probabilidad utilizando la función de distribución acumulada (cdf)
probabilidad_entre_250_y_350 = norm.cdf(Z_350) - norm.cdf(Z_250)

print(f'La probabilidad de ganar entre $250,00 y $350,00 es: {probabilidad_entre_250_y_350:.2%}')

# 2) Probabilidad entre $ 400,00 y $ 500,00

# Calcular los valores Z para los límites inferior y superior
Z_400 = (400.00 - media) / desviacion_estandar
Z_500 = (500.00 - media) / desviacion_estandar

# Calcular la probabilidad utilizando la función de distribución acumulada (cdf)
probabilidad_entre_400_y_500 = norm.cdf(Z_500) - norm.cdf(Z_400)

print(f'La probabilidad de ganar entre $400,00 y $500,00 es: {probabilidad_entre_400_y_500:.2%}')

La probabilidad de ganar entre $250,00 y $350,00 es: 68.27%
La probabilidad de ganar entre $400,00 y $500,00 es: 2.27%


### Problema C - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img006.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [94]:
# primero calculamos desde el limete inferior hata el 1.90 y luego restamos para transformarlo a lo que falta
media = 1.70
desviacion_estandar=0.1

In [95]:
Z = (1.9 - media) / desviacion_estandar
Z


1.9999999999999996

### Solución 1 - Utilizando la tabla

In [77]:
probabilidad = 0.9767
# probablidad hasta el 1.90 desde limite inferior hasta el 1.90

np.float64(0.9772498680518208)

In [96]:
probabilidad = 1- 0.9767
probabilidad
# probabilidad

0.023299999999999987

### Solución 2 - Utilizando Scipy

In [97]:
probabilidad = 1 - norm.cdf(Z)
probabilidad

np.float64(0.02275013194817921)

In [99]:
# otra forma de calcularlo
norm.cdf(-Z)
# me da la forma de arriba de un punto especifico

np.float64(0.022750131948179216)

## **EJERCICIOS**

In [101]:
# Datos del problema
media = 720
desviacion_estandar = 30

# 1) Probabilidad entre 650 y 750 días

# Calcular los valores Z para los límites inferior y superior
Z_650 = (650 - media) / desviacion_estandar
Z_750 = (750 - media) / desviacion_estandar

# Calcular la probabilidad utilizando la función de distribución acumulada (cdf)
probabilidad_entre_650_y_750 = norm.cdf(Z_750) - norm.cdf(Z_650)

print(f'La probabilidad de que una lámpara dure entre 650 y 750 días es: {probabilidad_entre_650_y_750:.2%}')

# 2) Probabilidad más de 800 días

# Calcular el valor Z para 800 días
Z_800 = (800 - media) / desviacion_estandar

# Calcular la probabilidad utilizando la función de supervivencia (sf)
probabilidad_mas_de_800 = norm.sf(Z_800)

print(f'La probabilidad de que una lámpara dure más de 800 días es: {probabilidad_mas_de_800:.2%}')

# 3) Probabilidad menos de 700 días

# Calcular el valor Z para 700 días
Z_700 = (700 - media) / desviacion_estandar

# Calcular la probabilidad utilizando la función de distribución acumulada (cdf)
probabilidad_menos_de_700 = norm.cdf(Z_700)

print(f'La probabilidad de que una lámpara dure menos de 700 días es: {probabilidad_menos_de_700:.2%}')

La probabilidad de que una lámpara dure entre 650 y 750 días es: 83.15%
La probabilidad de que una lámpara dure más de 800 días es: 0.38%
La probabilidad de que una lámpara dure menos de 700 días es: 25.25%


In [103]:
datos = [1.96, 2.15, -0.78, 0.59]
datos

[1.96, 2.15, -0.78, 0.59]

In [105]:
for i in datos:
    print(norm.cdf(i))


0.9750021048517795
0.9842223926089095
0.21769543758573312
0.7224046752465351


In [106]:
from scipy.stats import norm

# Item A
probabilidad = norm.cdf(1.96)
print(f'{probabilidad:0.4f}')

# Item B
probabilidad = 1 - norm.cdf(2.15)
# o -> probabilidad = norm.sf(2.15)
print(f'{probabilidad:0.4f}')

# Item C
probabilidad = norm.cdf(-0.78)
print(f'{probabilidad:0.4f}')

# Item D
probabilidad = 1 - norm.cdf(0.59)
# o -> probabilidad = norm.sf(0.59)
print(f'{probabilidad:0.4f}')

0.9750
0.0158
0.2177
0.2776


# <font color=green>3 MUESTREO</font>
***

## <font color=green>3.1 Población y Muestra</font>
***

### Población
Conjunto de todos los elementos de interés de un estudio. Varios elementos pueden componer una población, por ejemplo: personas, edades, alturas, automóviles, etc.

En cuanto al tamaño, las poblaciones pueden ser limitadas (poblaciones finitas) o ilimitadas (poblaciones infinitas).

### Poblaciones finitas

Permiten el recuento de sus elementos. Los ejemplos incluyen el número de empleados en una empresa, el número de estudiantes en una escuela, etc.

### Poblaciones infinitas

No es posible contar sus elementos. Como ejemplos tenemos el número de porciones que se pueden extraer del agua del mar para su análisis, la temperatura medida en cada punto de un territorio, etc.

<font color=red>Cuando se pueden contar los elementos de una población, pero presentando una cantidad muy grande, se supone que la población es infinita.</font>

### Muestra
Subconjunto representativo de la población.

Los atributos numéricos de una población, como su media, varianza y desviación estándar, se conocen como **parámetros**. El enfoque principal de la inferencia estadística es precisamente generar estimaciones y probar hipótesis sobre parámetros poblacionales utilizando información de la muestra.

## <font color=green>3.2 ¿Cuándo usar una muestra?</font>
***

### Poblaciones infinitas

El estudio nunca llegaría a su fin. No es posible investigar todos los elementos de la población.

### Pruebas destructivas

Estudios donde los elementos evaluados son totalmente consumidos o destruidos. Ejemplo: pruebas de vida útil, pruebas de seguridad en accidentes de tráfico.

### Resultados rápidos

Investigaciones que necesitan más agilidad en la difusión. Ejemplo: encuestas de opinión, encuestas sobre problemas de salud pública.

### Altos precios

Cuando la población es finita pero muy grande, el costo de un censo puede hacer inviable el proceso.

## <font color=green>3.3 Muestreo Aleatorio Simple</font>
***

Es una de las principales formas de extraer una muestra de una población. El requisito fundamental de este tipo de enfoque es que cada elemento de la población tenga las mismas posibilidades de ser seleccionado para formar parte de la muestra.

In [110]:
# Volvemos a cargar el DataFrame datos, ya que la variable fue sobreescrita más adelante en el notebook
datos = pd.read_csv('/content/datos.csv', encoding='latin1')
datos.shape[0]
# observamos el tamaño total de filas que tiene el banco de datos

76840

In [111]:
datos.Ingreso.mean()
# calculamos la media de ingreso

np.float64(2000.3831988547631)

In [120]:
muestra = datos.sample(n = 1000, random_state= 101)
muestra.shape[0]
# selecionamos nuestra muestra, en esete caso la cantidad y dar una semilla para reproducir mismos datos

1000

In [121]:
muestra.Ingreso.mean()
# obtenemos otra media, tenemos diferencia en la original

np.float64(1998.783)

In [122]:
# Debemos selecionarlar los datos represenattivos y asegurarnos de eso
datos.Sexo.value_counts(normalize=True) * 100


Sexo
0    69.299844
1    30.700156
Name: proportion, dtype: float64

In [124]:
muestra.Sexo.value_counts(normalize=True) * 100
# cuando se seleciona un variable o muestra aletorio se crea un margen de error deñ 0.5 %

Sexo
0    70.6
1    29.4
Name: proportion, dtype: float64

## <font color=green>3.4 Muestreo Estratificado</font>
***

Es una mejora del proceso de muestreo aleatorio simple. En este método, se propone dividir la población en subgrupos de elementos con características similares, es decir, grupos más homogéneos. Con estos subgrupos separados, la técnica de muestreo aleatorio simple dentro de cada subgrupo se aplica individualmente.

In [125]:
# Realizando muestreo estratificado por la columna 'Sexo'
# Se agrupa el DataFrame 'datos' por la columna 'Sexo'
# Se aplica una función lambda a cada grupo:
#   - Se toma una muestra aleatoria (sample) del grupo.
#   - La fracción (frac) del grupo que se toma como muestra es 0.1 (10%).
#   - random_state=101 asegura que la muestra sea reproducible.
# Se concatena verticalmente (axis=0) las muestras obtenidas de cada grupo para formar la muestra estratificada final.
muestra_estratificada = datos.groupby('Sexo').apply(lambda x: x.sample(frac=0.1, random_state=101))

# Mostrar el tamaño de la muestra estratificada
print("Tamaño de la muestra estratificada:")
print(muestra_estratificada.shape[0])

# Mostrar la distribución de la variable 'Sexo' en la muestra estratificada
print("\nDistribución de 'Sexo' en la muestra estratificada:")
print(muestra_estratificada.Sexo.value_counts(normalize=True) * 100)

# Mostrar la distribución de la variable 'Sexo' en la población original para comparar
print("\nDistribución de 'Sexo' en la población original:")
print(datos.Sexo.value_counts(normalize=True) * 100)

Tamaño de la muestra estratificada:
7684

Distribución de 'Sexo' en la muestra estratificada:
Sexo
0    69.299844
1    30.700156
Name: proportion, dtype: float64

Distribución de 'Sexo' en la población original:
Sexo
0    69.299844
1    30.700156
Name: proportion, dtype: float64


/tmp/ipython-input-62018710.py:8: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  muestra_estratificada = datos.groupby('Sexo').apply(lambda x: x.sample(frac=0.1, random_state=101))


## <font color=green>3.5 Muestreo por Conglomerados</font>
***

También tiene como objetivo mejorar el criterio de muestreo aleatorio simple. En el muestreo por conglomerados también se crean subgrupos, pero no serán homogéneos como en el muestreo estratificado. En el muestreo por conglomerados los subgrupos serán heterogéneos, donde luego se aplicará un muestreo aleatorio simple o estratificado.

Un ejemplo muy común de la aplicación de este tipo de técnica es la división de la población en grupos territoriales, donde los elementos investigados tendrán características bastante variadas.

In [126]:
# Realizando muestreo por conglomerados por la columna 'Ciudad'

# 1. Identificar los conglomerados (ciudades)
conglomerados = datos['Ciudad'].unique()

# 2. Seleccionar aleatoriamente un número de conglomerados
# En este ejemplo, seleccionaremos 5 ciudades al azar
num_conglomerados_a_seleccionar = 5
conglomerados_seleccionados = np.random.choice(conglomerados, num_conglomerados_a_seleccionar, replace=False)

# 3. Filtrar el DataFrame original para incluir solo los datos de los conglomerados seleccionados
muestra_conglomerados = datos[datos['Ciudad'].isin(conglomerados_seleccionados)]

# Mostrar el tamaño de la muestra por conglomerados
print("Tamaño de la muestra por conglomerados:")
print(muestra_conglomerados.shape[0])

# Mostrar la distribución de la variable 'Ciudad' en la muestra por conglomerados
print("\nDistribución de 'Ciudad' en la muestra por conglomerados:")
print(muestra_conglomerados['Ciudad'].value_counts(normalize=True) * 100)

# Mostrar la distribución de la variable 'Ciudad' en la población original para comparar (opcional)
print("\nDistribución de 'Ciudad' en la población original:")
print(datos['Ciudad'].value_counts(normalize=True) * 100)

Tamaño de la muestra por conglomerados:
14520

Distribución de 'Ciudad' en la muestra por conglomerados:
Ciudad
43    43.539945
23    23.133609
53    14.146006
32    10.406336
25     8.774105
Name: proportion, dtype: float64

Distribución de 'Ciudad' en la población original:
Ciudad
35    11.119209
31    10.002603
43     8.227486
29     7.440135
33     7.230609
15     5.789953
41     5.668922
26     4.971369
52     4.526288
23     4.371421
42     2.960698
13     2.870901
53     2.673087
51     2.342530
21     2.325612
11     2.000260
32     1.966424
50     1.874024
17     1.699636
28     1.674909
25     1.657991
22     1.576002
24     1.266268
12     1.219417
27     1.175169
14     0.702759
16     0.666320
Name: proportion, dtype: float64


# <font color=green>4 ESTIMACIÓN</font>
***

## <font color='red'>Problema </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una distribución normal **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un **intervalo de confianza para la media de la población** asumiendo un **nivel de significancia del 5%**.

---

Es la forma de hacer suposiciones generalizadas sobre los parámetros de una población a partir de información de una muestra.

- **Los parámetros** son los atributos numéricos de una población, como la media, la desviación estándar, etc.

- **Estimación** es el valor obtenido para un parámetro dado a partir de los datos de una muestra de la población.

## <font color=green>4.1 Teorema central del límite</font>
***

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

La desviación estándar de las medias muestrales se conoce como **error estándar de la media**

### Entendiendo el Teorema Central del Límite

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.2 Los niveles de confianza y significación</font>
***

El **nivel de confianza** ($1 - \alpha$) representa la probabilidad de que la estimación sea correcta. Además, el **nivel de significancia** ($\alpha$) expresa la probabilidad de error de la estimación.

El **nivel de confianza** representa el grado de confiabilidad del resultado estimado dentro de un cierto rango. Cuando establecemos un **nivel de confianza del 95%** en una encuesta, por ejemplo, estamos asumiendo que existe una probabilidad del 95% de que los resultados de la encuesta representen bien la realidad, es decir, sean correctos.

El **nivel de confianza** de una estimación se puede obtener del área bajo la curva normal como se muestra en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img007.png' width='450px'>


## <font color=green>4.3 Error inferencial</font>
***

El **error inferencial** se define por la **desviación estándar de las medias muestrales** $\sigma_\bar{x}$ y por el **nivel de confianza** determinado para el proceso.

# $$e = z \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.4 Intervalos de confianza</font>
***

### Intervalo de confianza para la media poblacional

#### Con desviación estándar poblacional conocida

## $$\mu = \bar{x} \pm z\frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar de población desconocida

## $$\mu = \bar{x} \pm z\frac{s}{\sqrt{n}}$$

## <font color='blue'>Ejemplo: </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una población normal con **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un intervalo de confianza para el **promedio de la población** asumiendo un **nivel de significancia del 5%**.

### Media muestral

### Nivel de significancia ($\alpha$)

### Nivel de confianza ($1 - \alpha$)

### Obteniendo $z$

### Obteniendo $z$

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img008.png' width='450px'>

### Valores de $z$ para los niveles de confianza más utilizados

|Nivel de<br>confianza|Valor del área bajo<br>la curva normal| $z$ |
|:----------------:|:---------------------------------:|:---:|
|90%               |0.95                               |1.645|
|95%               |0.975                              |1.96 |
|99%               |0.995                              |2.575|

### Obteniendo $\sigma_\bar{x}$

### Obteniendo $e$

### Solución 1 - Calculando el intervalo de confianza para la media

### Solución 2 - Calculando el intervalo de confianza para la media

# <font color=green>5 CÁLCULO DEL TAMAÑO DE LA MUESTRA</font>
***

## <font color='red'>Problema </font>

Estamos estudiando los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

## <font color=green>5.1 Variables cuantitativas y población infinita</font>
***

# $$e = z \frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar conocida

## $$n = \left(z\frac{\sigma}{e}\right)^2$$

#### Con desviación estándar desconocida

## $$n = \left(z\frac{s}{e}\right)^2$$

Donde:

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar de la muestra

$e$ = error inferencial

### <font color='red'>Comentarios</font>

1. La desviación estándar ($\sigma$ o $s$) y el error ($e$) deben estar en la misma unidad de medida.

2. Cuando el error ($e$) se representa en términos porcentuales, debe interpretarse como un porcentaje relacionado con el promedio.

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando los ingresos mensuales de los jefes de hogar. Nuestro supervisor determinó que el **error máximo en relación a la media es de $\$$ 100,00** pesos. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **$\$$ 3.323,39** pesos. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

### Obteniendo $\sigma$

### Obteniendo $e$

### Obteniendo $n$

---
---

## <font color='red'>Problema</font>

En un lote de **10 000 latas** de gaseosa, se realizó una muestra aleatoria simple de **100 latas** y se obtuvo que la **desviación estándar muestral del contenido de las latas es igual a 12 ml**. El fabricante estipula un **error máximo en la media poblacional de solo 5 ml**. Para garantizar un **nivel de confianza del 95%**, ¿qué tamaño de muestra se debe seleccionar para este estudio?

## <font color=green>5.2 Variables cuantitativas y población finita</font>
***

#### Con desviación estándar conocida

## $$n = \frac{z^2 \sigma^2 N}{z^2 \sigma^2 + e^2(N-1)}$$

#### Con desviación estándar desconocida

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

Donde:

$N$ = tamaño de la población

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar muestral

$e$ = error inferencial

## <font color='blue'>Ejemplo: Industria de refrescos</font>

En un lote de **10 000 latas** de gaseosa, se realizó una muestra aleatoria simple de **100 latas** y se obtuvo que la **desviación estándar muestral del contenido de las latas es igual a 12 ml**. El fabricante estipula un **error máximo en la media poblacional de solo 5 ml**. Para garantizar un **nivel de confianza del 95%**, ¿qué tamaño de muestra se debe seleccionar para este estudio?

### Obteniendo $N$

### Obteniendo $z$

### Obteniendo $s$

### Obteniendo $e$

### Obteniendo $n$

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

# <font color=green>6 FIJANDO EL CONTENIDO</font>
***

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando **los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil**. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79** y que la **media poblacional** es **R$\$$ 1.426,54**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio? ¿Cuál es el intervalo de confianza para la media considerando el tamaño de muestra obtenido?

### Construyendo el conjunto de datos según lo especificado por el problema

### Calcular el tamaño de la muestra

### Calcular el intervalo de confianza para la media

### Realización de una prueba gráfica